Adding textual input and output functionality.

WJCIV opened the pull request.

WJCIV committed:
Attempt to allow single record text to be read from files.

Changes limited (for now) to FileSelector.java.

WJCIV committed:
String input for files mostly implemented.

This fills in a lot of gaps in the implementation. Adds the binary/text
boolean to the file config. Adds support for text on accepts other than
the first. Adds cleanup to the BufferedReaders. Adds support for
character sets.

WJCIV committed:
Fixed some bad copy-paste with character sets in FileConfig.

WJCIV committed:
Syntax fixes for reading from text.

It compiles now. You can read text based input, and the records will be
put into the queue as byte[]. There's still something not right about
reading them back out.

WJCIV committed:
Added a TextFileSource.

Also added a TODO in FileSelector.

WJCIV committed:
Fixed the byte representation when setting a String in the Schema.

The Schema requires the length of the String precede the actual content.
This adds the functionality to put that information in there.

WJCIV committed:
Added Schema parser.

This allows the user to switch between String and byte[] for a record. A
default parser is supplied.

WJCIV committed:
Added a Schema parser interface and a separate Text Source output.

This extends the ability to read from text as the user can write how to
parse it into the Schema rather than forcing them into a single text
field. The output allows text output according to the same parser along
with new line delimiters. There's something odd with the output right
now - all the classes seem to be working, but I'm getting a completely
empty output file.

WJCIV committed:
Removed unused variable.

Turns out leaving it in allowed serialization because I probably forgot
to put the new code on the master, so the two sides of the connection
had different versions. With both sides running this code the
serialization works.

WJCIV committed:
Added a flush after each write to file and a FIXME comment about it.

It isn't as efficient to flush each write as to flush when the buffer
fills up and once at the very end, but this works for now.

raulcf commented:
This is pretty much ready to merge. Couple of minor comments:
- Feel free to refactor the new branching in FileSelector if(textsource) into private methods to improve readability.
- Name variables and classes according to Java conventions (for consistency with the rest of the codebase). A variable is textSource and not textsource and a class (regardless if static or private) always starts with a capital letter.

Other than that, I think as soon as the problem with flushing output data is fixed this can be merged.
Thanks!

WJCIV committed:
Cleanup and refactoring.

Moved some of the text input stuff in FileSelector into separate
functions for readability. Fixed the casing on some variables/classes.
Removed some unused imports.

WJCIV commented:
I think the flush problem is best fixed by adding either flush() or close() (or both) to the OBuffer interface. As it is now there's no way to indicate we are finished with an output stream, so there is no way to know when to flush. Do you have a preference for which of those functions to add?

raulcf commented:
Adding those functions to OBuffer interface is a temporal (valid) solution. Please proceed that way and also open an issue to revisit this. In particular, flush should be triggered when an underlying buffer gets full and when the query processing is stopped. Thanks!

WJCIV committed:
Added output buffer flushing in a more proper manner.

Added a flush function to the OBuffer interface and all of the known
implementing classes. Added a call to that flush function in the
shutdown hook. This avoids the need to flush after every write.
